scripts/leapfrog/elasticsearch_upgrade.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## Shell Opts ----------------------------------------------------------------
+set -e -u -x
+set -o pipefail
+
+# Check upgrade flag
+if [[ -n "$UPGRADE_ELASTICSEARCH" ]]; then
+  if [[ "$UPGRADE_ELASTICSEARCH" == "NO" ]]; then
+    echo 'Please make sure you want to upgrade Elasticsearch.'
+    exit
+  fi
+else
+  echo "Please preset env var before leap."
+  exit
+fi
+
+# TO-DO
+# Update pip.conf on Elasticsearch container
+pushd /opt/rpc-openstack/rpcd/playbooks
+openstack-ansible ../../scripts/leapfrog/playbooks/elasticsearch-postleap-pip-upgrade.yml
+popd
+
+# Stop logstash service on Logstash container
+pushd /opt/rpc-openstack/rpcd/playbooks
+openstack-ansible ../../scripts/leapfrog/playbooks/logstash-stop.yml
+popd
+
+# Run elasticsearch upgrade playbook
+pushd /opt/rpc-openstack/rpcd/playbooks
+openstack-ansible -e 'logging_upgrade=true' --tags 'reindex-wrapper,elasticsearch-upgrade' elasticsearch.yml
+popd
+
+# Restore logstash service on Logstash container
+pushd /opt/rpc-openstack/rpcd/playbooks
+openstack-ansible ../../scripts/leapfrog/playbooks/logstash-start.yml
+popd
```

Comment:
Is there a reason for the multiple `pushd`/`popd` lines here? Could we just put all of these lines together?

Reply:
Just want to make things uniform, but I can make them look simpler.